packages/workspace/src/browser/workspace-storage-service.ts

```diff
@@ -21,18 +22,24 @@ export class WorkspaceStorageService implements StorageService {
     private initialized: Promise<void>;
     protected storageService: StorageService;
 
-    constructor( @inject(WorkspaceService) protected workspaceService: WorkspaceService,
-        @inject(ILogger) protected logger: ILogger) {
-        this.initialized = this.workspaceService.root.then(stat => {
-            if (stat) {
-                this.prefix = stat.uri;
-            } else {
-                this.prefix = '_global_';
-            }
-        });
+    @inject(WorkspaceService)
+    protected workspaceService: WorkspaceService;
+
+    @inject(ILogger)
+    protected logger: ILogger;
+
+    constructor() {
         this.storageService = new LocalStorageService(this.logger);
     }
 
+    @postConstruct()
+    protected async init(): Promise<void> {
+        const statFile = await this.workspaceService.root;
+        const workspace = statFile ? new URI(statFile.uri).path : '_global_';
+        this.prefix = `${window.location.pathname}:${workspace}`;
+        this.initialized = Promise.resolve();
```

Comment:
It is not the same as before and can fail at runtime with `undefined` error.

Reply:
because `window` can be undefined ?

Reply:
`initialized` is not assigned and has `undefined` value at https://github.com/theia-ide/theia/pull/1748/files#diff-ae502ec06cf26d830ece3e21e6ec038eR22

It gets assigned async only after a workspace root is resolved if in the meantime someone tries to access it he will get an error.

Reply:
Good call, going to fix this, thank you.